activerecord/lib/active_record/associations/singular_association.rb

```diff
@@ -34,7 +34,7 @@ def force_reload_reader
 
       private
         def scope_for_create
-          super.except!(klass.primary_key)
+          super.except!(*Array(klass.primary_key))
```

Comment:
Long-term when we will be doing some internal cleanups I hope it will look like 
```suggestion
          super.except!(*klass.array_primary_key)
```

Where `array_primary_key` will be an internal concept representing most likely memoized value of the `primary_key` that is always an array so we can avoid unnecessary allocations. 

Until then I'm not against an explicit wrap as this is what we have been doing in other places and we will address it at once